crates/bevy_ecs/src/query/access.rs

```diff
@@ -387,6 +393,54 @@ impl<T: SparseSetIndex> From<FilteredAccess<T>> for FilteredAccessSet<T> {
     }
 }
 
+/// Records how two accesses conflict with each other
+#[derive(Debug, PartialEq)]
+pub enum AccessConflict {
+    /// Conflict is for all indices
+    All,
+    /// Conflict is only for a subset of indices
+    Individual(FixedBitSet),
+}
+
+impl AccessConflict {
+    fn add(&mut self, other: &Self) {
+        match (self, other) {
+            (s, AccessConflict::All) => {
+                *s = AccessConflict::All;
+            }
+            (AccessConflict::Individual(this), AccessConflict::Individual(other)) => {
+                this.extend(other.ones());
+            }
+            _ => {}
+        }
+    }
+
+    pub(crate) fn is_empty(&self) -> bool {
+        match self {
+            AccessConflict::All => false,
+            AccessConflict::Individual(set) => set.is_empty(),
+        }
+    }
+}
+
+impl From<FixedBitSet> for AccessConflict {
+    fn from(value: FixedBitSet) -> Self {
+        Self::Individual(value)
+    }
+}
+
+impl<T: SparseSetIndex> From<Vec<T>> for AccessConflict {
+    fn from(value: Vec<T>) -> Self {
+        Self::Individual(value.iter().map(T::sparse_set_index).collect())
+    }
+}
+
+impl Default for AccessConflict {
+    fn default() -> Self {
+        AccessConflict::Individual(FixedBitSet::new())
+    }
+}
+
```

Comment:
nit: i would rather we called this `AccessConflict::empty` or even `AccessConflict::nothing` for correctness. i'm not sure `default` conveys at fist glance that there are no conflicts. `AccessConflict::default` could just as reasonably return `AcessConflict::All` if we were being more conservative here.